libbeat/reader/readfile/line_test.go

```diff
@@ -418,7 +418,7 @@ func TestBufferSize(t *testing.T) {
 	bufferSize := 10
 
 	in := ioutil.NopCloser(strings.NewReader(strings.Join(lines, "")))
-	reader, err := NewLineReader(in, Config{codec, bufferSize, AutoLineTerminator, 1024})
+	reader, err := NewLineReader(in, Config{codec, bufferSize, AutoLineTerminator, 1024, false})
```

Comment:
could you add a test here where `CollectOnEOF` is true?  I know it gets tested in s3_objects_test.go, but if something happens to break it, it would be nice to have a test in the same directory.

Reply:
> could you add a test here where `CollectOnEOF` is true?

that was good! because I indeed spotted a bug where we were cutting the last `len(nl)` bytes from the returned line